hooks/boot/01-kafo-hook-extensions.rb

```diff
@@ -139,6 +139,18 @@ def execute_command(command, do_say, do_log)
   def remote_host?(hostname)
     !['localhost', '127.0.0.1', `hostname`.strip].include?(hostname)
   end
+
+  def needs_postgresql_upgrade?(new_version)
+    File.read('/var/lib/pgsql/data/PG_VERSION').chomp.to_i < new_version.to_i
```

Comment:
Should we deal with the file not existing?
```suggestion
    File.read('/var/lib/pgsql/data/PG_VERSION').chomp.to_i < new_version.to_i
  rescue Errno::ENOENT
    false
```

Reply:
You could argue this never gets executed if `local_postgresql?` is false, and if it is true that file really should exist.

OTOH, better safe than sorry :)